Core/include/Acts/Geometry/DetectorElementBase.hpp

```diff
@@ -31,23 +32,35 @@ class Surface;
 
 class DetectorElementBase {
  public:
-  DetectorElementBase() = default;
-  virtual ~DetectorElementBase() = default;
+  /// Construct a detector element. It will have ownership of the surface
+  DetectorElementBase(std::shared_ptr<Surface>&& surface) : m_surface(surface) {
```

Comment:
```suggestion
  explicit DetectorElementBase(std::shared_ptr<Surface>&& surface) : m_surface(surface) {
```

also
1. does the rvalue give us any benefit? looks like we could just use a value
2. this is a breaking change right? should we schedule that for v37?
3. should we make assumptions about the ownership? require that the shared pointer is not used anywhere else or enforce that with a `unique_ptr`?

Reply:
1. Its more a hint to the caller, but not strictly necessary
2. I would think that is definitively breaking.
3. So the issue I encountered with `unique_ptr` is that `std::shared_ptr<...>::unique` is actually removed from C++20 because of thread safety issues... We could assume that geometry construction is always done in a single thread however.